tools/extract_features.py

```diff
@@ -19,25 +19,26 @@
 
 
 def extract_caption_t5():
-    t5 = T5Embedder(device="cuda", local_cache=True, cache_dir='data/t5_ckpts')
-    t5_save_root = 'data/data_toy/caption_feature_wmask'
-    t5_save_dir = t5_save_root
+    t5 = T5Embedder(device="cuda", local_cache=True, cache_dir=f'{args.pretrained_models_dir}/t5_ckpts')
+    t5_save_dir = args.t5_save_root
     os.makedirs(t5_save_dir, exist_ok=True)
     captions = set()
 
-    train_data_json = json.load(open('data/data_toy/data_info.json', 'r'))
+
+    train_data_json = json.load(open(args.json_path, 'r'))
     train_data = train_data_json[args.start_index: args.end_index]
+    images_extension = "." + train_data[0]['path'].rsplit('.', 1)[-1]
```

Comment:
better to use parser for extension, in case, there is no extension in data['path']